server/src/main/java/org/elasticsearch/threadpool/ThreadPoolInfo.java

```diff
@@ -23,11 +22,11 @@ public class ThreadPoolInfo implements ReportingService.Info, Iterable<ThreadPoo
     private final List<ThreadPool.Info> infos;
 
     public ThreadPoolInfo(List<ThreadPool.Info> infos) {
```

Comment:
Does this constructor actually need to be public? Then we can just keep the semantics inside the class, and don't need to wrap inside the constructor

Reply:
I guess it could be package private but either way it hardly matters, just went for this change to always have the same type here an make it clear we never have `null` in this list.